### ISSUE TITLE
Can't import runfiles module from rules_go

### ISSUE THREAD

**Issue by ted-xie** [bazelbuild/rules_go#3469]
### What version of rules_go are you using?

v0.38.1

### What version of gazelle are you using?

v0.28.0

### What version of Bazel are you using?

6.0.0

### Does this issue reproduce with the latest releases of all the above?

The last green bazel release (e79de51b91263b33ced77f0a749a1856972510d1) also fails in the same way.

### What operating system and processor architecture are you using?

Debian-based Linux distro. x86-64.

### What did you do?

Tried to import `"github.com/bazelbuild/rules_go/go/runfiles"` in a go_test with `"@io_bazel_rules_go//go/runfiles:runfiles"` in the test's `deps` attribute. I would have expected the rules_go toolchain to append `github.com/bazelbuild/rules_go/go/runfiles` to the import path already, since the runfiles module co-exists with the rest of the bazel go toolchain.

```
git clone https://github.com/ted-xie/rules_go_runfiles
cd rules_go_runfiles
bazel test //src:a_test
```

### What did you expect to see?

I would have expected the test/build to complete successfully.

### What did you see instead?

```
compilepkg: missing strict dependencies:
        /usr/local/google/home/tedx/.cache/bazel/_bazel_tedx/8fdee1a60d976347f04b72ccefe7568d/sandbox/linux-sandbox/1/execroot/rules_go_runfiles_repro/src/a_test.go: import of "github.com/bazelbuild/rules_go/go/runfiles"
No dependencies were provided.
Check that imports in Go sources match importpath attributes in deps.
Target //src:a_test failed to build
```

**Comment by ted-xie** [comment]
@fmeum

**Comment by fmeum** [comment]
This seems to be WAI: If I delete the `BUILD` file in `src` and run `gazelle`, the generated test target builds and tests successfully. It includes a dependency on `@io_bazel_rules_go//go/runfiles:go_default_library`, which is the expected target providing the runfiles library.

@ted-xie Could you explain why you would have expected the `runfiles` library to be accessible without a `deps` entry?

**Comment by ted-xie** [comment]
Ah, so what had happened was that I did originally have `@io_bazel_rules_go//go/runfiles:go_default_library` in `deps` of `a_test`, but after I ran gazelle, that dep somehow got moved to a new target called `src_test`. I did not follow the common convention of naming directories the same way as packages/modules, which is part of the confusion.

Let me try integrating this into my real project, and if it's another silly mistake like this then I'll close this issue as WAI.
